test/test_pipeline/test_losses.py

```diff
@@ -32,29 +31,24 @@ def test_get_name_error():
     name = 'BCELoss'
     with pytest.raises(ValueError, match=r"Invalid name entered for task [a-z]+_[a-z]+, "):
-        get_loss_instance(dataset_properties, name)
+        get_loss(dataset_properties, name)
 
 
 @pytest.mark.parametrize('weighted', [True, False])
-def test_losses(weighted):
-    list_properties = [{'task_type': 'tabular_classification', 'output_type': 'multiclass'},
-                       {'task_type': 'tabular_classification', 'output_type': 'binary'},
-                       {'task_type': 'tabular_regression', 'output_type': 'continuous'}]
-    pred_cross_entropy = torch.randn(4, 4, requires_grad=True)
-    list_predictions = [pred_cross_entropy, torch.empty(4).random_(2), torch.randn(4)]
-    list_names = [None, 'BCEWithLogitsLoss', None]
-    list_targets = [torch.empty(4, dtype=torch.long).random_(4), torch.empty(4).random_(2), torch.randn(4)]
-    labels = [torch.empty(20, dtype=torch.long).random_(4), torch.empty(12, dtype=torch.long).random_(2), None]
-    for dataset_properties, pred, target, name, label in zip(list_properties, list_predictions,
-                                                             list_targets, list_names, labels):
-        loss = get_loss_instance(dataset_properties=dataset_properties, name=name)
-        weights = None
-        if bool(weighted) and 'classification' in dataset_properties['task_type']:
-            strategy = get_loss_weight_strategy(output_type=STRING_TO_OUTPUT_TYPES[dataset_properties['output_type']])
-            weights = strategy(y=label)
-            weights = torch.from_numpy(weights)
-            weights = weights.type(torch.FloatTensor)
-            kwargs = {'pos_weight': weights} if 'binary' in dataset_properties['output_type'] else {'weight': weights}
-        loss = loss() if weights is None else loss(**kwargs)
-        score = loss(pred, target)
-        assert isinstance(score, torch.Tensor)
+@pytest.mark.parametrize('loss_details', ['loss_cross_entropy_multiclass',
+                                          'loss_cross_entropy_binary',
+                                          'loss_bce',
+                                          'loss_mse'], indirect=True)
+def test_losses(weighted, loss_details):
+    dataset_properties, predictions, name, targets, labels = loss_details
+    loss = get_loss(dataset_properties=dataset_properties, name=name)
+    weights = None
+    if bool(weighted) and 'classification' in dataset_properties['task_type']:
+        strategy = get_loss_weight_strategy(loss)
+        weights = strategy(y=labels)
+        weights = torch.from_numpy(weights)
+        weights = weights.type(torch.FloatTensor)
+        kwargs = {'pos_weight': weights} if loss.__name__ == 'BCEWithLogitsLoss' else {'weight': weights}
+    loss = loss() if weights is None else loss(**kwargs)
+    score = loss(predictions, targets)
+    assert isinstance(score, torch.Tensor)
```

Comment:
Just to make sure nothing is broken, can you check the size of this score and you actually are providing a fixed target... I think the value in this case is deterministic?

Reply:
in this case no it is not deterministic, I didn't add the seeding. But do you think it will be better to add the seed for all np, torch at the top of the contest file?

Reply:
I have updated the test_losses in a way that we don't need to set the seed for this test as per our discussion today